### PR TITLE
feat(cake-prediction): Add Cake Prediction

### PR DESCRIPTION
**Pull request by chefilip**

### CODE REVIEW

File: src/views/Predictions/components/RoundCard/SetPositionCard.tsx [hunk 1]
```diff
@@ -73,19 +76,38 @@ const getValueAsEthersBn = (value: string) => {
...
+
+  const { isVaultApproved, setLastUpdated } = useCakeApprovalStatus(token.symbol === 'CAKE' ? predictionsAddress : null)
+  const { handleApprove, pendingTx } = useCakeApprove(
+    setLastUpdated,
+    predictionsAddress,
+    t('You can now start prediction'),
+  )
+
+  // BNB prediction doesn't need approval
+  const doesCakeApprovePrediction = token.symbol === 'BNB' || isVaultApproved
```
Comment (memoyil) [inline src/views/Predictions/components/RoundCard/SetPositionCard.tsx:108]: Why don't we use useApproveConfirmTransaction hook?

Comment (chefilip) [reply src/views/Predictions/components/RoundCard/SetPositionCard.tsx:108]: AFAIK, `useApproveConfirmTransaction` is low-level hook which is used for more generic cases. If we use `useApproveConfirmTransaction` for such specific cases, the duplication would increase.

File: src/views/Predictions/context/config.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import addresses from 'config/constants/contracts'
+
+import { GRAPH_API_PREDICTION_CAKE, GRAPH_API_PREDICTION } from 'config/constants/endpoints'
+import { getAddress } from 'utils/addressHelpers'
+import tokens from 'config/constants/tokens'
+
+export default {
+  BNB: {
+    address: getAddress(addresses.predictions),
+    api: GRAPH_API_PREDICTION,
```
Comment (memoyil) [inline src/views/Predictions/context/config.ts:10]: name can have bnb suffix

Comment (chefilip) [reply src/views/Predictions/context/config.ts:10]: Good idea. I will do it after release because it requires a env config on Vercel.

**Review by chef-jojo — DISMISSED** [review]

**Pull request merged**